rlp/main.cpp

```diff
@@ -256,14 +257,20 @@ int main(int argc, char** argv)
         prefs.escapeAll = true;
 
     bytes in;
-    if (inputFile == "--")
-        for (int i = cin.get(); i != -1; i = cin.get())
-            in.push_back((byte)i);
-    else if (boost::filesystem::is_regular_file(inputFile))
-        in = contents(inputFile);
-    else
-        in = asBytes(inputFile);
-
+    try
+    {
+        if (inputFile == "--")
+            for (int i = cin.get(); i != -1; i = cin.get())
+                in.push_back((byte)i);
+        else if (boost::filesystem::is_regular_file(inputFile))
+            in = contents(inputFile);
+    }
+    catch(const boost::filesystem::filesystem_error& _e)
+    {
+        if(_e.code() == boost::system::errc::errc_t::filename_too_long)
```

Comment:
It's ok to handle all file system errors the same way (errors from `contents`, too), not only `filename_too_long`